File: RoomSense-Pi-Device/src/Entities/ActuationCommand.py
from Entities.Enums.Actuator import Actuator

class ActuationCommand:
    def __init__(
            self,
            actuationCommandId = 0,
            actuator = None,
            status = None,
            createdAt = None,
            executedAt = None
        ):
            self.actuationCommandId = actuationCommandId
            self.actuator = actuator
            self.status = status
            self.createdAt = createdAt
            self.executedAt = executedAt

    def Print(self):
        created_str = self.createdAt.strftime("%Y-%m-%d %H:%M:%S") if self.createdAt else "None"
        executed_str = self.executedAt.strftime("%Y-%m-%d %H:%M:%S") if self.executedAt else "None"

        print(f"""
            Actuator Command:
            ID: {self.actuationCommandId}
            Actuator Device: {self.actuator}
            Command Status: {self.status}
            Created At: {created_str}
            Executed At: {executed_str}
        """)

File: RoomSense-Web-Client/src/main/java/com/k00282713/RoomSense_Web_Client/entities/ActuationCommand.java
package com.k00282713.RoomSense_Web_Client.entities;

import com.k00282713.RoomSense_Web_Client.entities.enums.Actuator;
import com.k00282713.RoomSense_Web_Client.entities.enums.Status;
import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "actuation_command")
public class ActuationCommand
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer actuationCommandId;

    @Enumerated(EnumType.STRING)
    private Actuator actuator;

    @Enumerated(EnumType.STRING)
    private Status status;

    private LocalDateTime createdAt;
    private LocalDateTime  executedAt;

    public ActuationCommand() { }
    public ActuationCommand(
        Integer actuationCommandId,
        Actuator actuator,
        Status status,
        LocalDateTime createdAt,
        LocalDateTime executedAt
    )
    {
        this.actuationCommandId = actuationCommandId;
        this.actuator = actuator;
        this.status = status;
        this.createdAt = createdAt;
        this.executedAt = executedAt;
    }

    public Integer getActuationCommandId()
    {
        return actuationCommandId;
    }

    public void setActuationCommandId(Integer actuationCommandId)
    {
        this.actuationCommandId = actuationCommandId;
    }

    public Actuator getActuator()
    {
        return actuator;
    }

    public void setActuator(Actuator actuator)
    {
        this.actuator = actuator;
    }

    public Status getStatus()
    {
        return status;
    }

    public void setStatus(Status status)
    {
        this.status = status;
    }

    public LocalDateTime getCreatedAt()
    {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt)
    {
        this.createdAt = createdAt;
    }

    public LocalDateTime getExecutedAt()
    {
        return executedAt;
    }

    public void setExecutedAt(LocalDateTime executedAt)
    {
        this.executedAt = executedAt;
    }
}
